src/zenml/cli/utils.py

```diff
@@ -1437,15 +1437,28 @@ def inner_decorator(func: F) -> F:
         data_type_descriptors = set()
         for k, v in filter_model.__fields__.items():
             if k not in filter_model.CLI_EXCLUDE_FIELDS:
-                options.append(
-                    click.option(
-                        f"--{k}",
-                        type=str,
-                        default=v.default,
-                        required=False,
-                        help=create_filter_help_text(filter_model, k),
+                # if k is metadata, add option with multiple values
+                if k == "metadata_values":
+                    options.append(
+                        click.option(
+                            "--metadata",
+                            "-m",
+                            type=str,
+                            multiple=True,
+                            required=False,
+                            help=create_filter_help_text(filter_model, k),
```

Comment:
The field name should be `labels`, not `metadata`.

Second thing: this is a generic decorator, so adding exceptions for attribute names here is not going to scale well. Just imagine 10 different if clauses piling up here, one for every special case.

The correct way to do it is as follows:
1. add the `labels` CLI argument to the "list stack component" command, like I did here with `label`: https://github.com/zenml-io/zenml/blob/af927f2bc1399b27fdbd2438e4aa6df64511442b/src/zenml/cli/service_connectors.py#L372-L379
2. inhibit the `label` field in the StackComponentFilterModel, like I did with the ServiceConnectorFilterModel here: https://github.com/zenml-io/zenml/blob/af927f2bc1399b27fdbd2438e4aa6df64511442b/src/zenml/models/service_connector_models.py#L938 . This tells the generic decorator to not process this field.
3. parse and add the `label` values to the filter model in the CLI command implementation, like I did here with labels: https://github.com/zenml-io/zenml/blob/af927f2bc1399b27fdbd2438e4aa6df64511442b/src/zenml/cli/service_connectors.py#L849-L854

You should maybe run some tests to make sure this works.

Reply:
This is beautiful. Thanks for the fix, I would however, remove this altogether like we did with the other PR. This is because we currently don't support listing by labels due to the fact that we store labels as bytes and filtering by that is a bit complex.
I'd however come back to this later and use this design when I do.

Reply:
Yeah, label filtering is a pain, I agree. I'm on board with leaving it out until we get it right in the back-end.